samza-core/src/test/java/org/apache/samza/diagnostics/TestDiagnosticsStreamMessage.java

```diff
@@ -0,0 +1,133 @@
+    List<DiagnosticsManager.ProcessorStopEvent> stopEventList = new ArrayList<>();
+    stopEventList.add(new DiagnosticsManager.ProcessorStopEvent("0", executionEnvContainerId, hostname, 101));
+    return stopEventList;
+  }
+
+  public static Map<String, ContainerModel> getSampleContainerModels() {
+    Map<String, ContainerModel> containerModels = new HashMap<>();
+    Map<TaskName, TaskModel> tasks = new HashMap<>();
+
+    Set<SystemStreamPartition> sspsForTask1 = new HashSet<>();
+    sspsForTask1.add(new SystemStreamPartition("kafka", "test-stream", new Partition(0)));
+    tasks.put(new TaskName("Partition 0"), new TaskModel(new TaskName("Partition 0"), sspsForTask1, new Partition(0)));
+
+    Set<SystemStreamPartition> sspsForTask2 = new HashSet<>();
+    sspsForTask2.add(new SystemStreamPartition("kafka", "test-stream", new Partition(1)));
+    tasks.put(new TaskName("Partition 1"), new TaskModel(new TaskName("Partition 1"), sspsForTask2, new Partition(1)));
+
+    containerModels.put("0", new ContainerModel("0", tasks));
+    return containerModels;
+  }
+
+  @Test
+  /**
+   * Tests basic operations on {@link DiagnosticsStreamMessage}.
+   */ public void basicTest() {
+
+    DiagnosticsStreamMessage diagnosticsStreamMessage = getDiagnosticsStreamMessage();
+    Collection<DiagnosticsExceptionEvent> exceptionEventList = getExceptionList();
+    diagnosticsStreamMessage.addDiagnosticsExceptionEvents(exceptionEventList);
+    diagnosticsStreamMessage.addProcessorStopEvents(getProcessorStopEventList());
+    diagnosticsStreamMessage.addContainerModels(getSampleContainerModels());
+
+    Assert.assertEquals(1024, diagnosticsStreamMessage.getContainerMb());
+    Assert.assertEquals(2, diagnosticsStreamMessage.getContainerNumCores());
+    Assert.assertEquals(3, diagnosticsStreamMessage.getNumStoresWithChangelog());
+    Assert.assertTrue(exceptionEventList.equals(diagnosticsStreamMessage.getExceptionEvents()));
+    Assert.assertEquals(getSampleContainerModels(), diagnosticsStreamMessage.getContainerModels());
+    Assert.assertEquals(diagnosticsStreamMessage.getProcessorStopEvents(), getProcessorStopEventList());
+  }
+
+  @Test
+  /**
+   * Tests serialization and deserialization of a {@link DiagnosticsStreamMessage} using {@link MetricsSnapshotSerdeV2}.
+   * This serde is currently used by {@link DiagnosticsManager} to serde messages to the diagnostics stream.
+   */ public void serdeTest() {
+    DiagnosticsStreamMessage diagnosticsStreamMessage = getDiagnosticsStreamMessage();
+    Collection<DiagnosticsExceptionEvent> exceptionEventList = getExceptionList();
+    diagnosticsStreamMessage.addDiagnosticsExceptionEvents(exceptionEventList);
+    diagnosticsStreamMessage.addProcessorStopEvents(getProcessorStopEventList());
+    diagnosticsStreamMessage.addContainerModels(getSampleContainerModels());
+
+    byte[] serializedBytes = new MetricsSnapshotSerdeV2().toBytes(diagnosticsStreamMessage.convertToMetricsSnapshot());
```

Comment:
The API for `DiagnosticsStreamMessage` is `convertToMetricsSnapshot` and `convertToDiagnosticsStreamMessage`, so you shouldn't need to call `MetricsSnapshotSerdeV2` here.
In my opinion, this test should have the following flow:
1) `convertToMetricsSnapshot`
2) check that the resulting `MetricsSnapshot` is correct
3) `convertToDiagnosticsStreamMessage`
4) check that the resulting `DiagnosticsStreamMessage` matches the original